src/java/org/apache/cassandra/service/ReplicaFilteringProtection.java

```diff
@@ -289,6 +266,35 @@ public void close()
         };
     }
 
+    private void checkCachedRowThresholds()
+    {
+        if (rowsCached == cachedRowsFailThreshold + 1)
+        {
+            String message = String.format("Replica filtering protection has cached over %d rows during query %s. " +
```

Comment:
What we would recommend to someone hitting this would be, beside changing the thresholds, improving the sync between replicas and/or lowering the page size. Should we mention this here, in the yaml, or in the documentation?

Reply:
I'll try to expand the inline description in `cassandra.yaml` a bit, given the log message already refers the reader there.